resource_type.go

```diff
@@ -76,12 +82,34 @@ func (t ResourceType) validate(raw []byte) (ResourceAttributes, *errors.ScimErro
 			return ResourceAttributes{}, scimErr
 		}
 
-		attributes[extension.Schema.ID] = extensionAttributes
+		schemaAttributes[extension.Schema.ID] = extensionAttributes
+	}
+
+	attributes := make(map[string]interface{}, len(commonAttributes)+len(schemaAttributes))
+
+	for k, v := range schemaAttributes {
+		attributes[k] = v
+	}
+
+	for k, v := range commonAttributes {
+		attributes[k] = v
 	}
 
 	return attributes, nil
 }
 
+func (t ResourceType) validateCommonAttributes(m map[string]interface{}) (map[string]interface{}, *errors.ScimError) {
+	if eID, ok := m["externalId"]; ok {
```

Comment:
Q: should we introduce a constant for this key

Reply:
I would not be bad to create constants for all the common attributes.
https://github.com/elimity-com/scim/blob/f934e79b8e439cf9849023d8fd45dc6bd4a01660/resource_handler.go#L52
https://github.com/elimity-com/scim/blob/f934e79b8e439cf9849023d8fd45dc6bd4a01660/resource_handler.go#L77

Reply:
I've added them to the schema package